gr-analog/lib/squelch_base_ff_impl.cc

```diff
@@ -100,8 +119,10 @@ int squelch_base_ff_impl::general_work(int noutput_items,
             break;
 
         case ST_DECAY:
-            // FIXME: precalculate window for speed
-            d_envelope = 0.5 - std::cos(GR_M_PI * (--d_ramped) / d_ramp) / 2.0;
+            if (!d_window_table.empty()) {
+                d_envelope = d_window_table[--d_ramped];
+            }
+
             if (d_ramped == 0.0) {
```

Comment:
huh, this code is comparing `int d_ramped` with `double` 0.0; can you fix that while you're at it?